contracts/FlowIDTableStaking.cdc

```diff
@@ -1710,16 +1710,35 @@ pub contract FlowIDTableStaking {
     pub fun getProposedNodeIDs(): [String] {
```

Comment:
I am confused about the intended separation of responsibilities between this function and `endStakingAuction`. In particular, with the new changes, this function seems to duplicate some checks from [`removeInvalidNodes`](https://github.com/onflow/flow-core-contracts/pull/321/files#diff-65336be374bb3fc9ad7b822243e065a389d73e758c7c16223e52fc3181cea59bR978). 
- We added the `movesPendingList` in order to limit the number of nodes we would need to check when processing the end of the staking auction, however here we are still iterating through all nodes.
- We are checking the same approval status and stake here as we already did in `removeInvalidNodes`

Reply:
You're right about it being more computationally intensive, but I was trying to make it so we don't have to do two upgrades. `FlowEpoch` calls this function to get the list of nodes for the next epoch and I wanted to make the return value of this function accurate to represent all the proposed nodes. If we wanted to upgrade the `FlowEpoch` contract also, we could just make the `endStakingAuction` return the proposed node IDs and use that for `EpochSetup` so we only have to iterate through it once, but we have to get this changes upgraded on testnet today, so I didn't want to make too many changes in case it takes too long to test and upgrade. What do you think?

Reply:
yeah, I think we should include this in a future upgrade. We just need to get this method working properly today, and I don't have time tonight to fully figure out the best way to do it and test it. Sorry about that. I'll create an issue.

Reply:
> we could just make the endStakingAuction return the proposed node IDs and use that for EpochSetup

👍 Agree with this suggestion, and getting rid of `getProposedList`. The proposed list used to mean, "if the staking auction ends right now, this represents what the identity table would be" -- it was useful as a way to read the tentative identity table between epoch phases, in addition to being used to read the final, real identity table by `FlowEpoch`. But with random slot selection, an accurate tentative identity table doesn't really exist.

Reply:
I created an issue for this too (#342) and we'll include it in the next upgrade